scipy/optimize/_lsq/least_squares.py

```diff
@@ -390,13 +405,12 @@ def least_squares(
         no effect with ``loss='linear'``, but for other `loss` values it is
         of crucial importance.
     max_nfev : None or int, optional
-        Maximum number of function evaluations before the termination.
-        If None (default), the value is chosen automatically:
+        For all methods this parameter controls the maximum number of function
+        evaluations separate to those used in numerical approximation of the jacobian.
+        If None (default), the value is chosen automatically as 100 * n.
 
-        * For 'trf' and 'dogbox' : 100 * n.
-        * For 'lm' :  100 * n if `jac` is callable and 100 * n * (n + 1)
-          otherwise (because 'lm' counts function calls in Jacobian
-          estimation).
+        .. versionchanged:: 1.16.0
+            The default for the 'lm' method is changed to 100 * n.
```

Comment:
```suggestion
            The default for the 'lm' method is changed to 100 * n, for both a callable and
            a numerically estimated jacobian. Previously the default when using an estimated
            jacobian was 100 * n * (n + 1), because the method included evaluations used in
            the estimation. 
```